src/script/cpp_api/s_env.cpp

```diff
@@ -154,6 +153,59 @@ void ScriptApiEnv::initializeEnvironment(ServerEnvironment *env)
 			// removes value, keeps key for next iteration
 			lua_pop(L, 1);
 		}
+	} else {
+		lua_pop(L, 1);
+		FATAL_ERROR("FATAL ERROR: core.registered_abms was not a lua table, as expected.");
+	}
```

Comment:
Okay did it below for the new code, but didnt want to turn this into a cleanup. The PR has over 500 changed lines already.